app/javascript/controllers/dump_controller.js

```diff
@@ -0,0 +1,66 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["list", "template"]
+  static values = { type: String }
+
+  connect() {
+    this.getDumpData(this.element, this.typeValue);
+  }
+
+  getDumpData(target, type) {
+    fetch('https://s3-us-west-2.amazonaws.com/rubygems-dumps/?prefix=production/public_' + type)
+      .then(response => response.text())
+      .then(data => {
+        const parser = new DOMParser();
+        const xml = parser.parseFromString(data, "application/xml");
+        const files = this.parseS3Listing(xml);
+        this.render(files);
+      })
+      .catch(error => {
+        console.error(error);
+      });
+  }
+
+  parseS3Listing(xml) {
+    const contents = Array.from(xml.getElementsByTagName('Contents'));
+    return contents.map(item => {
+      return {
+        Key: item.getElementsByTagName('Key')[0].textContent,
+        LastModified: item.getElementsByTagName('LastModified')[0].textContent,
+        Size: item.getElementsByTagName('Size')[0].textContent,
+        StorageClass: item.getElementsByTagName('StorageClass')[0].textContent
+      };
+    });
+  }
+
+  render(files) {
+    files
+      .sort((a, b) => Date.parse(b.LastModified) - Date.parse(a.LastModified))
+      .filter(item => 'STANDARD' === item.StorageClass)
```

Comment:
Nit: filter before sort is faster

Reply:
Good catch. I didn't do much more than copy the original javascript (removing jquery).